### PR TITLE
Feature: upgradeable domains, auto 404 fallback on website domains

### PR DESCRIPTION
**Pull request by philogicae**

### CODE REVIEW

File: src/domain/domain.ts [hunk 1]
```diff
@@ -212,18 +223,19 @@ export class DomainManager implements EntityManager<Domain, AddDomain> {
     throwOnCollision = true,
   ): Promise<AddDomain[]> {
     domains = await DomainManager.addManySchema.parseAsync(domains)
+    if (!throwOnCollision) return domains
 
     const currentDomains = await this.getAll()
     const currentDomainSet = new Set<string>(currentDomains.map((d) => d.name))
 
-    if (!throwOnCollision) {
+    /* if (!throwOnCollision) {
```
Comment (amalcaraz) [inline src/domain/domain.ts:231]: The purpose of this is not signing domain messages for already existing domains. Why are we removing it?

File: src/domain/website.ts [hunk 2]
```diff
@@ -613,6 +613,7 @@ export class WebsiteManager implements EntityManager<Website, AddWebsite> {
               message_id: volumeId,
               type: cv.target,
               updated_at: new Date().toISOString(),
+              options: { catch_all_path: '/404.html' },
```
Comment (amalcaraz) [inline src/domain/website.ts:616]: Why are we handling domain aggregates from here? This should be managed by the domain class not by the website class. We should invoke the proper method of the domain class from here

File: src/hooks/pages/configure/useNewDomainPage.ts [hunk 3]
```diff
@@ -100,7 +101,10 @@ export function useNewDomainPage(): UseNewDomainPageReturn {
     setValue,
     formState: { errors },
   } = useForm({
-    defaultValues,
+    defaultValues: {
```
Comment (amalcaraz) [inline src/hooks/pages/configure/useNewDomainPage.ts:104]: calculate this in a useMemo. Cause we are creating a new object each time the component re renders

**Review by amalcaraz — CHANGES_REQUESTED** [review]

File: src/domain/executable.ts [hunk 4]
```diff
@@ -161,7 +161,7 @@ export abstract class Executable {
       ref,
     }))
 
-    return yield* this.domainManager.addSteps(parsedDomains, false)
+    return yield* this.domainManager.addSteps(parsedDomains, 'override')
```
Comment (amalcaraz) [inline src/domain/executable.ts:164]: shouldn't it be 'ignore' here to keep the old behaviour when we are not trying to update them?

Comment (philogicae) [reply src/domain/executable.ts]: Fixed

**Pull request merged**